src/components/pages/dashboard-pages/Notifications.jsx

```diff
@@ -1,10 +1,211 @@
+          (classInfo) =>
+            myClassIds.includes(classInfo._id) &&
+            Array.isArray(classInfo.applications) &&
+            classInfo.applications.length > 0
+        );
+
+        // Sort applications within each class
+        const classesWithSortedApplications =
+          sortClassesByApplicationDate(filteredClasses);
+
+        // Now sort the classes based on the earliest application date
+        const sortedClasses = sortClassesByEarliestApplicationDate(
+          classesWithSortedApplications
+        );
+
+        setClasses(sortedClasses);
+        setClassesError(null);
+      } catch (error) {
+        console.error('Error fetching classes data:', error);
+        setClassesError({
+          message: 'Failed to fetch classes. Please try again later.',
+        });
+      }
+    };
+
+    getTeacherClasses();
+  }, [userData]);
+
+  useEffect(() => {
+    if (userData.role !== 'teacher' || classes.length === 0) return;
+
+    const getStudentInfo = async () => {
+      try {
+        const token = userData.token;
+        const response = await getAllUsersInfo(token);
+        const allApplicants = response || [];
+        const myApplicantIds = classes.flatMap((classInfo) =>
+          classInfo.applications.map((application) => application.userId)
+        );
+
+        const filteredApplicants = allApplicants.filter((response) =>
+          myApplicantIds.includes(response._id)
+        );
+
+        setApplicants(filteredApplicants);
+        setApplicantsError(null);
+      } catch (error) {
+        console.error('Error fetching student data:', error);
+        setApplicantsError({
+          message: 'Failed to fetch students. Please try again later.',
+        });
+      }
+    };
+
+    getStudentInfo();
+  }, [classes, userData]);
+
+  const handleApprove = (classId, applicationId) => {
+    console.log('Approved:', classId, applicationId);
+  };
+
+  const handleDecline = (classId, applicationId) => {
+    console.log('Declined:', classId, applicationId);
+  };
+
+  if (userData.role !== 'teacher') {
+    return <p>Coming soon...</p>;
+  }
+
   return (
-    <>
+    <div className="container mt-10">
       <h1 className="text-red font-bold text-2xl sm:text-4xl font-spartan uppercase">
-        Notifications page - common
+        New applications:
       </h1>
-    </>
+      {classesError && <p className="text-red-500">{classesError.message}</p>}
+      <div className="flex flex-col gap-6 mt-10 m-5 md:m-10">
+        {classes.length > 0 ? (
+          classes.map((classInfo) => (
+            <div
+              key={classInfo._id}
+              className="border rounded-lg border-gray p-5 bg-pureWhite"
+            >
+              <h2 className="font-roboto font-medium text-2xl mb-4">
+                {classInfo.classTitle}
+              </h2>
+              <hr className="my-4 border-gray" />
+              {classInfo.applications.map((application) => {
```

Comment:
Technically sometimes is better approach instead of map in jsx with long return to move this function in separate state or component. It is optional here and does not require to refactor code

Reply:
Thank you, Anna!
I plan to rewrite teacher notifications (new it is applications) in separate PR